cirrus/google_cloud/manager.py

```diff
@@ -1075,6 +1121,53 @@ def _get_proxy_group_name_for_user(user_id, username):
     return str(username).strip().replace(" ", "-") + "-" + str(user_id)
 
 
+def _get_proxy_group_service_account_id_for_user(user_id, username):
+    """
+    Return a valid service account id based on user_id and username
+
+    Currently Google enforces the following:
+        6-30 characters
+        Must match: [a-z][a-z\d\-]*[a-z\d]
+
+    Args:
+        user_id (str): User's uuid
+        username (str): user's name
+
+    Returns:
+        str: service account id
+    """
+    username = str(username).strip().replace(" ", "-")
```

Comment:
If the priority is to have the user ID in the account ID and the username might not fit, and it's fine to have it first, I think this function could be shortened to:
```python
account_id = user_id + '-' + username
# truncate to 30
account_id = account_id[:30]
# pad to at least 6
account_id += (6 - len(account_id)) * '-'
```

Reply:
unfortunately the account id can't start with a number so the user id can't go first (which resulted in this gross logic, I agree). Google's validation regex : [a-z][a-z\d\-]*[a-z\d]

Reply:
ah right, sorry. I wasn't reading carefully enough

Reply:
I'm sure there's a cleaner way to do what I'm doing here, but the requirement still has to be 
1) user_id is fully available
2) must match google's validation regex

Reply:
```python
# Truncate username so full account ID is at most 30 characters.
full_account_id_length = len(username) + len(user_id) + 1
chars_to_drop = full_account_id_length - 30
truncated_username = username[:-chars_to_drop]
account_id = truncated_username + '-' + user_id
# Pad account ID to at least 6 chars long.
account_id += (6 - len(account_id)) * '-'
```

Reply:
I like it, thanks! changed in af860e3026a9724d207b351e2bdacb0dc4a8612f